components/script/dom/node.rs

```diff
@@ -483,6 +483,24 @@ impl Node {
         self.flags.set(flags);
     }
 
+    // FIXME(emilio): This and the function below should move to Element.
+    pub fn note_dirty_descendants(&self) {
+        debug_assert!(self.is_in_doc());
+
+        // FIXME(emilio): Is there a safe way to do this without the overhead of
+        // rooting? I'm pretty sure this cannot GC!
+        let mut current = Some(DomRoot::from_ref(self));
+
+        while let Some(node) = current.take() {
```

Comment:
There's no good answer for the rooting question, unfortunately. As for this code, let's use:
```rust
for node in self.ancestors() {
    if node.get_flag(...) {
        return;
    }
    node.set_flag(...);
}
```

Reply:
I changed it to inclusive_ancestors, which is what it is intended to do.